shared/src/commonMain/kotlin/io.newm.shared/di/Koin.kt

```diff
@@ -79,4 +86,31 @@ class LoginUseCaseFactory() : KoinComponent {
     fun loginUseCase(): LoginUseCase {
         return loginUseCase
     }
+}
+
+class SignupUseCaseFactory() : KoinComponent {
```

Comment:
Seems like we will have a lot of repetitions with all these factories... What if we create a `ClientCaseFactory` and provide all the cases there? Maybe we can move that to a separate file.